<HTML>
<HEAD>
   <TITLE>tinker files</TITLE>
</HEAD>

<body bgcolor="#FFFFFF">
<BODY>

<CENTER><H2>Files and Subprograms used by Tinker</H2></CENTER>
<HR>
<A HREF="http://dasher.wustl.edu/tinker">TINKER</A> is a collection of programs particularly 
well suited for the construction and manipulation of peptides and oligonucleotides. The
programs are the result of many years of work by <A HREF=http://dasher.wustl.edu>Jay Ponder</A>
and his research group. The program suite operates through the action of various subprograms 
on a set of files sharing the same principal name (such as <I>testz1</I>), but carrying variable 
endings depending on the particular purpose and content of the file. The most important files 
are:<BR><BR>

<FONT COLOR="#000FF">the cartesian coordinate file (.xyz)</FONT><BR>
This file contains the cartesian coordinates of the system. The atom labels used in this file depend on the
force field chosen in the keyword file (see below).<BR><BR>  

<FONT COLOR="#000FF">the internal coordinate file (.int)</FONT><BR>
This file contains the structure of the system in internal coordinates. As for the xyz-coordinate file
the atom labels depend on the force field chosen in the keyword file.<BR><BR>

<FONT COLOR="#000FF">the sequence file (.seq)</FONT><BR>
This file contains the sequence of the peptide (using the three-letter code) or oligonucleotide.<BR><BR>

<FONT COLOR="#000FF">the keyword file (.key)</FONT><BR>
This file contains additional information such as the force field parameters that should be used during
geometry optimization or molecular dynamics calculations of the system.<BR><BR>

One general note with respect to the files used by <FONT COLOR="#000FF">Tinker</FONT> should be made on 
file updating. In order
to avoid overwriting of existing files, <FONT COLOR="#000FF">Tinker</FONT> generates new files in a 
systematic way in case older files
should exist. Should, for example, the coordinate file <I>testz1.xyz</I> exist at the beginning of a geometry
optimization, a new file named <I>testz1.xyz_2</I> will be generated containing the optimized coordinates. 
Repeated execution of the optimization program with increasingly accurate convergence criteria will
generate the files <I>testz1.xyz_3</I>, <I>testz1.xyz_4</I> and so on. On startup the <FONT COLOR="#000FF">Tinker</FONT> 
programs will therefore search for the file with the highest version number and use this file 
for input.<BR><BR>

The programs of the <FONT COLOR="#000FF">Tinker</FONT> suite are all located in the directory
/usr/local/tinker/bin. In order to start the programs from the tcsh command line interface, you
should add the following line to your .cshrc file:<BR><BR>

<FONT COLOR="#000FF">
set path = ( /usr/local/tinker  /usr/local/tinker/bin $path)
</FONT><BR><BR>

Also, in order to organize the Java environment used by <FONT COLOR="#000FF">Tinker</FONT> and
<FONT COLOR="#000FF">FFE</FONT> appropriately, check whether either the file <I>libjvm.so</I> exists 
in the directory /usr/lib, or whether a link has been set to point to the corresponding location
in the <FONT COLOR="#000FF">Tinker</FONT> directory such as:<BR><BR>

<FONT COLOR="#000FF">
libjvm.so -> /usr/local/tinker/jre/lib/i386/client/libjvm.so
</FONT><BR><BR>

The most important subprograms of the program suite are:<BR><BR>

<H4>structure generation and manipulation</H4>
<FONT COLOR="#000FF">protein</FONT><BR>
This subprogram builds up peptide chains and generates the corresponding coordinate files (.xyz and .int)
and the sequence file (.seq).<BR><BR> 

<FONT COLOR="#000FF">archive</FONT><BR>
This program takes a number of single coordinate files (in xyz format) and produces one
large file containing all the structures. The program can also be used to extract single
structures out of existing archives.<BR><BR>

<FONT COLOR="#000FF">intxyz</FONT><BR>
A program for the interconversion of internal and cartesian coordinates.
If invoked with <FONT COLOR="#000FF">intxyz testz1</FONT> the program reads the internal
coordinates from file testz1.int and writes out the cartesian coordinates to file testz1.xyz.
The conversion of cartesian to internal coordinates can analogously be performed using the
program <FONT COLOR="#000FF">xyzint</FONT>.
<BR><BR>

<H4>structure optimization</H4>
<FONT COLOR="#000FF">minimize</FONT><BR>
This is a gradient minimizer requiring only the energy and the gradient in order to optimize the
structure of the system. It is recommend for the preliminary minimization of structures generated by the
<FONT COLOR="#000FF">protein</FONT> subprogram. The first argument given to the program is the principal 
name of the system under investigation (such as <I>testz1</I>) and the second argument is the convergence 
criterion in kcal/mol/angstroms. A criterion of 0.1 may be adequate for preliminary geometry optimization.<BR><BR>

<FONT COLOR="#000FF">newton</FONT><BR>
This is a geometry optimizer based on a Newton-Raphson algorithm (requiring energy, gradient, and
second derivative information) that can also be coached to optimize conformational transition states. 
As input the program accepts the principal name of the system under investigation, additional
information for the particular variant of the algorithm and for preconditioning the system (both
of which can be directed with the argument <I>a</I> to assume default values), and finally the convergence 
criterion (in kcal/mol/angstroms).<BR><BR>

<FONT COLOR="#000FF">optimize</FONT><BR>
This is a gradient minimizer operating in cartesian coordinate space and requiring calculation of the 
energy, the gradients, and an approximate second derivative matrix. The first argument given to the 
program is the principal name of the system under investigation and the second argument is the convergence 
criterion in kcal/mol/angstrom.<BR><BR>

<H4>conformational search</H4>
<FONT COLOR="#000FF">scan</FONT><BR>
After initial geometry optimization of the input structure (supplied in a xyz-coordinates file) the 
program searches the conformational space in a systematic fashion by following low energy torsional
modes from one conformational isomer to the next one. After each "hopping" step, the program performs
a local geometry optiomization of the structure. The conformational minima found in this way are included
in a list of conformers, whose low energy torsional modes are followed in order to find new conformational
isomers. In case all modes of all known conformers have been tested without finding new minima, the search
is ended. The program accepts as arguments the principal name of the system under study (in order to
locate the corresponding xyz-coordinates file), information on the torsional angles included in the 
search process (0 chooses automatic selection), the number of search directions during the local
search (default is 5), the energy threshold for local minima (default is 100.0 kcal/mol), and the convergence
criterion for local geometry optimizations (default is 0.0001 kcal/mol/angstrom).<BR><BR>

<FONT COLOR="#000FF">sniffer</FONT><BR>
This program implements the sniffer global trajectory method for searching for global minima. 
As input the program accepts the principal name of the system under study in order to locate
the xyz-coordinate file used as the seed structure. The subsequent arguments accepted by the program
are the number of trial structures (default is 100), the target energy of the system (default is 0.0),
and the convergence criterion per atom (default is 1.0).<BR><BR> 

<H4>structure characterization</H4>
<FONT COLOR="#000FF">vibrate</FONT><BR>
This program calculates the vibrational frequency spectrum of the structure described in the input (.xyz) file.
As input the program takes the principal file name of the system as well as the number of the vibration, for
which displacements are to be written to the output file. The list of vibration numbers is terminated by
0.<BR><BR>

<FONT COLOR="#000FF"><A HREF="tink5.html">superpose</A></FONT><BR>
This program predicts the best possible superposition of two structures <I>a</I> and <I>b</I>
and calculates the final residual difference d<SUB>ab</SUB> (rms distance) between the structures. 

<HR>
<PRE>
<TT>last changes: 18.11.2005, HZ
questions & comments to: <A HREF="mailto:zipse@cup.uni-muenchen.de">zipse@cup.uni-muenchen.de</A>
</TT>
</PRE> 


</BODY>
</HTML>
<BR><BR>


